src/org/jetbrains/plugins/scala/codeInspection/functionExpressions/UnnecessaryPartialFunctionQuickFix.scala

```diff
@@ -0,0 +1,54 @@
+package org.jetbrains.plugins.scala.codeInspection.functionExpressions
+
+import com.intellij.openapi.project.Project
+import org.jetbrains.plugins.scala.codeInspection.functionExpressions.UnnecessaryPartialFunctionQuickFix._
+import org.jetbrains.plugins.scala.codeInspection.{AbstractFixOnPsiElement, InspectionBundle}
+import org.jetbrains.plugins.scala.lang.psi.ScalaPsiUtil
+import org.jetbrains.plugins.scala.lang.psi.api.base.patterns.{ScCaseClause, ScReferencePattern, ScWildcardPattern}
+import org.jetbrains.plugins.scala.lang.psi.api.expr.{ScBlock, ScBlockExpr}
+import org.jetbrains.plugins.scala.lang.psi.impl.ScalaPsiElementFactory
+
+object UnnecessaryPartialFunctionQuickFix {
+  val hint = InspectionBundle.message("convert.to.anonymous.function")
+}
+
+class UnnecessaryPartialFunctionQuickFix(caseClause: ScCaseClause, expression: ScBlockExpr)
+  extends AbstractFixOnPsiElement(hint, caseClause) {
+
+  override def doApplyFix(project: Project): Unit = {
```

Comment:
I understand that. It is about lifecycle of a quickfix. If another case clause is added and quickfix is invoked before the file gets rehighlighted, it will throw a MatchError.